convert/convert.go

```diff
@@ -35,9 +35,45 @@ type builder struct {
 	imports []*proto.Import
 }
 
-// Run converts a proto file to a gunk file, saving the file in the same folder
-// as the proto file.
-func Run(path string, overwrite bool) error {
+// Run converts proto files or folders to gunk files, saving the files in
+// the same folder as the proto file.
+func Run(paths []string, overwrite bool) error {
+	for _, path := range paths {
+		if err := run(path, overwrite); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func run(path string, overwrite bool) error {
+	fi, err := os.Stat(path)
```

Comment:
you might want to make this logic simpler. for example, the way `go build` does it, it builds files if the first one has a `.go` extension. that way, `go build ./foo.go` errors if `foo.go` is something very weird like a directory, and they save the extra stat work. you likely also want to error on cases like `gunk convert ./some-folder` where `some-folder` is actually a proto file.

Reply:
I don't quite follow sorry? I also don't see how this logic could be simpler, it is finding all the .proto files in a directory or using the path if it is a .proto file?

Reply:
Right now you do:

```
if !IsDir(path) {
    return convertFile(...)
}
// convert all files within dir that end in ".proto"
```

What I was suggesting was to instead do:

```
if HasSuffix(path, ".proto") {
    return convertFile(...)
}
// convert all files within dir that end in ".proto"
```

This is the same thing that `go build` does. For example, try doing `go build foo.txt` if the file exists. It won't error saying that the file has bad syntax - it will say that it can't find the package, because the argument isn't treated as a Go file.

Reply:
What if `path` is a directory in this case? The error in the case where we take your approach is:

```
error: unable to parse proto file "some.proto": go scanner error at <input> = read some.proto: is a directory
```

Where if we check that the `path` is a directory then we can correctly handle the case where someone is using a `example.proto` as a directory name.

Reply:
Yeah, Go doesn't support directories with `.go` extensions, and I don't think we should support that for gunk or proto directories either. In their case, they do catch that error to give a slightly better one:

```
$ mkdir foo.go
$ go build foo.go
foo.go is a directory, should be a Go file
```

So I don't think this would save you the stat call, then :)

Reply:
Is there a reason not to support that though? I can't think of a good reason not to? It seems like if someone wants to name their directories like that, and it is trivial for us to handle, then should we handle this case?

Happy to remove it if there is a reason that this is a bad idea though :)

Reply:
I've always assumed that the reason why they keep the logic string-based is because when one sees `go build foo.go` or `go build ./bar`, it's always clear whether we're building a single file or a directory/package. It's not clear if it depends on what's on disk. This is why I was initially saying "simpler logic".

Ultimately we've got bigger fish to fry, so feel free to go either way :)

Reply:
Ah, seems reasonable to me actually. If someone actually hits this issue we can always revisit the way we do this. I have added a check for `IsDir(path) && HasSuffix(path, ".proto")` to error on this case.